Cannot use slice as input

```
error[E0277]: the trait bound `&[u8; 3]: PyClass` is not satisfied
   --> src/lib.rs:94:1
    |
94  | / export_cpy!(
95  | |     mod navigator {
96  | |         enum AdcChannel {
97  | |             Ch0,
...   |
251 | |     }
252 | | );
    | |_^ the trait `PyClass` is not implemented for `&[u8; 3]`
```

--- a/example/cpp_project/main.cpp
+++ b/example/cpp_project/main.cpp
@@ -10,6 +10,8 @@ int main() {
     printf("Tire material: %s\n", materials[(unsigned int)tire.material]);
     printf("Tire size: %.2fw, %.2fh\n", tire.size.width, tire.size.height);
     printf("The tire have an aspect ratio of %.2f\n",wheel_size_aspect(tire.size.width, tire.size.height));
+    uint8_t values[3] = {77, 42, 69};
+    format_wheel_identifier(&values);
 
     func_with_no_return();
 

--- a/example/py_project/main.py
+++ b/example/py_project/main.py
@@ -4,5 +4,5 @@ tire = example.create_random_tire()
 print(f'Tire pressure: {tire.pressure:.2f}')
 print(f'Tire material: {tire.material}')
 print(f'Tire size: {tire.size.width:.2f} width, {tire.size.height:.2f} height')
-
+example.format_wheel_identifier([77, 42, 69])
 example.func_with_no_return()

--- a/example/src/lib.rs
+++ b/example/src/lib.rs
@@ -42,6 +42,14 @@ export_cpy!(
             }
         }
 
+        fn_c format_wheel_identifier(dimensions: &[u8; 3]) {
+            println!("Wheel identifier: {dimensions:?}");
+        }
+
+        fn_py format_wheel_identifier(dimensions: Vec<u8>) {
+            println!("Wheel identifier: {dimensions:?}");
+        }
+
         fn func_with_no_return() {
             println!("Yep, no returns");
         }

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -33,6 +33,14 @@ macro_rules! export_cpy {
         export_cpy!(@generate_function $name($($param : $ptype),*) $(-> $ret)? $body);
         export_cpy!(@process_item $($rest)*);
     };
+    (@process_item fn_c $name:ident($($param:ident : $ptype:ty),*) $(-> $ret:ty)? $body:block $($rest:tt)*) => {
+        export_cpy!(@generate_c_function $name($($param : $ptype),*) $(-> $ret)? $body);
+        export_cpy!(@process_item $($rest)*);
+    };
+    (@process_item fn_py $name:ident($($param:ident : $ptype:ty),*) $(-> $ret:ty)? $body:block $($rest:tt)*) => {
+        export_cpy!(@generate_py_function $name($($param : $ptype),*) $(-> $ret)? $body);
+        export_cpy!(@process_item $($rest)*);
+    };
 
     //(2) - This section defines the structure of the itens
     (@generate_enum $name:ident { $($variant:ident,)* }) => {
@@ -61,6 +69,18 @@ macro_rules! export_cpy {
         pub extern "C" fn $name($($arg: $arg_type),*) $(-> $ret)?
             $body
     };
+    (@generate_c_function $name:ident($($arg:ident: $arg_type:ty),*) $(-> $ret:ty)? $body:block) => {
+        #[no_mangle]
+        #[cfg(not(feature = "python"))]
+        pub extern "C" fn $name($($arg: $arg_type),*) $(-> $ret)?
+            $body
+    };
+    (@generate_py_function $name:ident($($arg:ident: $arg_type:ty),*) $(-> $ret:ty)? $body:block) => {
+        #[cfg(feature = "python")]
+        #[pyfunction]
+        pub fn $name($($arg: $arg_type),*) $(-> $ret)?
+            $body
+    };
 
     //(3) - This section defines the bindings to be exported to Python module
     (@add_py_binding $m:ident,) => {};
@@ -73,6 +93,13 @@ macro_rules! export_cpy {
         export_cpy!(@add_py_binding $m, $($rest)*);
     };
     (@add_py_binding $m:ident, fn $name:ident($($param:ident : $ptype:ty),*) $(-> $ret:ty)? $body:block $($rest:tt)*) => {
+        $m.add_wrapped(wrap_pyfunction!($name))?;
+        export_cpy!(@add_py_binding $m, $($rest)*);
+    };
+    (@add_py_binding $m:ident, fn_c $name:ident($($param:ident : $ptype:ty),*) $(-> $ret:ty)? $body:block $($rest:tt)*) => {
+        export_cpy!(@add_py_binding $m, $($rest)*);
+    };
+    (@add_py_binding $m:ident, fn_py $name:ident($($param:ident : $ptype:ty),*) $(-> $ret:ty)? $body:block $($rest:tt)*) => {
         $m.add_wrapped(wrap_pyfunction!($name))?;
         export_cpy!(@add_py_binding $m, $($rest)*);
     };

